### ISSUE TITLE
How should I set the epoch?

### ISSUE THREAD

**Issue by GA12WAINM** [QData/spacetimeformer#43]
Hello dear author,  I cloned your project code and tested it,  and found that the epochs cannot be set in the command,  and the exact location of the epochs cannot be found in the file.  I read the args-related part of the instruction set in the file(E.g：--batch_size --workers) and can't seem to find it. How did the author set the epochs? If possible,  could you point out how to modify the epochs?  Thank you very much if you would like to help.

**Comment by 1347919709** [comment]
Hello, have you found where to modify the epoch？ I am also plagued by this problem.

**Comment by GA12WAINM** [comment]
> 

Unfortunately, I can't find the corresponding epochs command setting location for the time being. I've run most of the tasks so far and found that it seems to use an automatic stopping mechanism that stops epochs when the loss value drops to a local minimum.

**Comment by GA12WAINM** [comment]
Unfortunately, I can't find the corresponding epochs command setting location for the time being. I've run most of the tasks so far and found that it seems to use an automatic stopping mechanism that stops epochs when the loss value drops to a local minimum.




KOˊ
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "QData/spacetimeformer"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月16日(星期一) 下午4:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [QData/spacetimeformer] How should I set the epoch? (Issue #43)





 
Hello, have you found where to modify the epoch？ I am also plagued by this problem.

**Comment by jakegrigsby** [comment]
Hi, we don't set an epoch number and instead use Early Stopping and a reduced learning rate to end the training process. The learning rate decreases when validation loss does not improve. If that doesn't seem to be helping then training process ends. This is a pretty fair way to set training schedules when comparing models with significantly different parameter counts and training times.

 If you wanted to get around this you could edit the `Trainer` object at the bottom of `train.py` to include the `max_epochs` flag. 
https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html
and then remove the EarlyStopping callback in `create_callbacks`

(I got email notifications for a few more comments on this issue but they're mysteriously gone. I've never seen that before.. not sure if that's a GitHub bug or if they were all deleted.)

**Comment by GA12WAINM** [comment]
Thank you very much for your reply in your busy schedule! According to your opinion, I adjusted the relevant pl.trainer section in the file [train.py], added the settings of the epoch section and disabled Early_stopping, and the test result is feasible.
